scripts/deploy/deploy.rb

```diff
@@ -42,13 +42,41 @@ def execute_steps(steps, step_index)
           'Continue from a specified step') do |t|
     @step_index = t.to_i
   end
+
+  opts.on('--dry-run', "Don't do a real deployment, but test what would happen if you did") do |t|
+      @is_dry_run = t
+  end
+
+  opts.on('--branch', "Branch to deploy from") do |t|
+      @branch = t
+  end
 end.parse!
 
 steps = [
+    # Prep for making changes
     method(:validate_version_number),
+    method(:ensure_clean_repo),
+    method(:pull_latest),
+    method(:switch_to_release_branch),
+
+    # Update version number
     method(:update_read_me),
     method(:update_stripe_sdk_version),
     method(:update_gradle_properties),
+    method(:create_version_bump_pr),
 ]
 
-execute_steps(steps, @step_index)
+begin
+    execute_steps(steps, @step_index)
+rescue
+    delete_release_branch()
```

Comment:
I think we'd ideally group these with the steps that created it. I'd hate to start at a later stage, only to have it fail and remove stuff I already thought was complete.

Reply:
good point! Fixed this. 

I added this because if the release branch already existed remotely, trying to push to it from my local branch would fail. So updated this to instead try to delete the branch before creating it, to ensure the local working copy is the only copy of that branch.